<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>BuiltinDirectoryConfigurationPage</class>
 <widget class="QWidget" name="BuiltinDirectoryConfigurationPage">
  <property name="windowTitle">
   <string>Builtin directory</string>
  </property>
  <layout class="QGridLayout" name="gridLayout" rowstretch="0,0,1,0,0" columnstretch="0,1">
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item row="0" column="0" colspan="2">
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Directory name</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="directoryName"/>
     </item>
    </layout>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="legacyLocalDataNetworkObjects">
     <property name="text">
      <string>Locations</string>
     </property>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QLabel" name="networkObjects">
     <property name="text">
      <string>Computers</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QTableWidget" name="locationTableWidget">
     <attribute name="horizontalHeaderVisible">
      <bool>false</bool>
     </attribute>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>true</bool>
     </attribute>
     <attribute name="verticalHeaderVisible">
      <bool>false</bool>
     </attribute>
     <column>
      <property name="text">
       <string>Locations</string>
      </property>
     </column>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QTableWidget" name="computerTableWidget">
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <attribute name="horizontalHeaderMinimumSectionSize">
      <number>150</number>
     </attribute>
     <attribute name="horizontalHeaderStretchLastSection">
      <bool>true</bool>
     </attribute>
     <attribute name="verticalHeaderVisible">
      <bool>false</bool>
     </attribute>
     <column>
      <property name="text">
       <string>Name</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Host address/IP</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>MAC address</string>
      </property>
     </column>
    </widget>
   </item>
   <item row="3" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QPushButton" name="addLocationButton">
       <property name="toolTip">
        <string>Add new location</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../../core/resources/core.qrc">
         <normaloff>:/core/list-add.png</normaloff>:/core/list-add.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="removeLocationButton">
       <property name="toolTip">
        <string>Remove selected location</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../../core/resources/core.qrc">
         <normaloff>:/core/edit-delete.png</normaloff>:/core/edit-delete.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item row="3" column="1">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="addComputerButton">
       <property name="toolTip">
        <string>Add new computer</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../../core/resources/core.qrc">
         <normaloff>:/core/list-add.png</normaloff>:/core/list-add.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="removeComputerButton">
       <property name="toolTip">
        <string>Remove selected computer</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../../core/resources/core.qrc">
         <normaloff>:/core/edit-delete.png</normaloff>:/core/edit-delete.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="importStudentsButton">
       <property name="toolTip">
        <string>Import students from CSV file</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../../core/resources/core.qrc">
         <normaloff>:/core/document-open.png</normaloff>:/core/document-open.png</iconset>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QLabel" name="label_3">
     <property name="font">
      <font>
       <italic>true</italic>
      </font>
     </property>
     <property name="text">
      <string>Importing CSV files is possible through the command line interface. For more information, see the &lt;a href=&quot;https://docs.veyon.io/en/latest/admin/cli.html#network-object-directory&quot;&gt;online documentation&lt;/a&gt;.</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../../core/resources/core.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>addComputerButton</sender>
   <signal>clicked()</signal>
   <receiver>BuiltinDirectoryConfigurationPage</receiver>
   <slot>addComputer()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>646</x>
     <y>528</y>
    </hint>
    <hint type="destinationlabel">
     <x>507</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeComputerButton</sender>
   <signal>clicked()</signal>
   <receiver>BuiltinDirectoryConfigurationPage</receiver>
   <slot>removeComputer()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>698</x>
     <y>528</y>
    </hint>
    <hint type="destinationlabel">
     <x>507</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>removeLocationButton</sender>
   <signal>clicked()</signal>
   <receiver>BuiltinDirectoryConfigurationPage</receiver>
   <slot>removeLocation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>528</y>
    </hint>
    <hint type="destinationlabel">
     <x>507</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addLocationButton</sender>
   <signal>clicked()</signal>
   <receiver>BuiltinDirectoryConfigurationPage</receiver>
   <slot>addLocation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>56</x>
     <y>528</y>
    </hint>
    <hint type="destinationlabel">
     <x>507</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>locationTableWidget</sender>
   <signal>cellChanged(int,int)</signal>
   <receiver>BuiltinDirectoryConfigurationPage</receiver>
   <slot>updateLocation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>364</y>
    </hint>
    <hint type="destinationlabel">
     <x>507</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>computerTableWidget</sender>
   <signal>cellChanged(int,int)</signal>
   <receiver>BuiltinDirectoryConfigurationPage</receiver>
   <slot>updateComputer()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>630</x>
     <y>364</y>
    </hint>
    <hint type="destinationlabel">
     <x>507</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>importStudentsButton</sender>
   <signal>clicked()</signal>
   <receiver>BuiltinDirectoryConfigurationPage</receiver>
   <slot>importStudents()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>750</x>
     <y>528</y>
    </hint>
    <hint type="destinationlabel">
     <x>507</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>addLocation()</slot>
  <slot>removeLocation()</slot>
  <slot>addComputer()</slot>
  <slot>removeComputer()</slot>
  <slot>updateLocation()</slot>
  <slot>updateComputer()</slot>
  <slot>importStudents()</slot>
 </slots>
</ui>
